### PR TITLE
Adding VIES VAT api

### PR DESCRIPTION
**Pull request by patka-123**

### CODE REVIEW

**Comment by markdingemanse** [comment]
src/ViesClient.php (blank_line_after_namespace) looks like there was a validation error. PR looks good to me otherwise

File: src/Vat.php [hunk 1]
```diff
@@ -5,28 +5,35 @@
...
+        ?ResolvesCountries $countryResolver = null,
+        ?ResolvesVatRates $vatRateResolver = null,
+        ?VerifiesVatNumbers $vatNumberVerifier = null
+    ) {
+        $this->countryResolver = $countryResolver ?? new Iso2();
+        $this->vatRateResolver = $vatRateResolver ?? new TaxesEuropeDatabaseClient();
+        $this->vatNumberVerifier = $vatNumberVerifier ?? new ViesClient();
     }
 
     public function validateVatNumber(string $vatNumber, string $countryCode): bool
```
Comment (markdingemanse) [inline src/Vat.php]: Could we change the name of this function to validateEuropeanVatNumber

Comment (markdingemanse) [reply src/Vat.php]: the naming atm would suggest i could also verify my 日本  vat number

**Review by markdingemanse — CHANGES_REQUESTED** [review]

File: src/Vat.php (hunk 1 above)
Comment (patka-123) [reply src/Vat.php]: Yessir!

File: README.md [hunk 2]
```diff
@@ -26,7 +26,7 @@
 ```php
 $vatServcie = new \SandwaveIo\Vat\Vat;
 
-$vatServcie->validateVatNumber("YOURVATNUMBERHERE"); // true
+$vatServcie->validateEuropeanVatNumber("YOURVATNUMBERHERE"); // true
```
Comment (ThomasRedpanda) [inline README.md]:
```suggestion
$vatServcie->validateEuropeanVatNumber("YOURVATNUMBERHERE", "COUNTRYCODE"); // true
```

**Review by ThomasRedpanda — APPROVED** [review]

**Review by FrontEndCoffee — APPROVED** [review]
If you add the case for vat number validation without country prefix like we talked about, you can merge this. Great work!

**Pull request merged**